src/test/ui/mismatched_types/closure-arg-count.stderr

```diff
@@ -23,23 +23,47 @@ error[E0308]: mismatched types
    = note: expected type `&{integer}`
               found type `(_, _)`
 
-error[E0593]: closure takes 1 argument but 2 arguments are required
+error[E0593]: closure is expected to take 2 arguments, but it takes 1 argument
   --> $DIR/closure-arg-count.rs:17:15
    |
 17 |     [1, 2, 3].sort_by(|(tuple, tuple2)| panic!());
-   |               ^^^^^^^ -------------------------- takes 1 argument
+   |               ^^^^^^^ ----------------- takes 1 argument
    |               |
    |               expected closure that takes 2 arguments
 
-error[E0593]: closure takes 0 arguments but 1 argument is required
+error[E0593]: closure is expected to take 1 argument, but it takes 0 arguments
   --> $DIR/closure-arg-count.rs:18:5
    |
 18 |     f(|| panic!());
-   |     ^ ----------- takes 0 arguments
+   |     ^ -- takes 0 arguments
    |     |
    |     expected closure that takes 1 argument
    |
    = note: required by `f`
 
-error: aborting due to 5 previous errors
+error[E0593]: closure is expected to take a single tuple as argument, but it takes 2 distinct arguments
+  --> $DIR/closure-arg-count.rs:20:53
+   |
+20 |     let _it = vec![1, 2, 3].into_iter().enumerate().map(|i, x| i);
+   |                                                     ^^^ ------ help: consider changing to: `|(i, x)|`
+   |                                                     |
+   |                                                     expected closure that takes 1 argument, a 2-tuple
```

Comment:
This is looking really good! I hope you don't mind me nitpicking some more though. =)

Reading this "expected closure that takes 1 argument, a 2-tuple" -- which I think I may even have initially suggested -- I see now that it is actually ambiguous. In particular, the intention was something like "that takes 1 argument, which is a 2-tuple", but it could also be parsed as a list, like "that takes 1 argument and a 2-tuple". In fact, this second interpretation is what I got this time, and I had to re-read it a few times to understand what was meant.

Therefore, I suggest:

"expected closure that takes a single tuple as argument"

which echoes the wording from above